lib/msprime.c

```diff
@@ -2651,18 +2658,33 @@ msp_gene_conversion_event(msp_t *self, label_id_t label)
     x = y->prev;
 
     /* generate tract length */
-    do {
-        tl = gsl_ran_exponential(self->rng, self->gc_tract_length);
```

Comment:
Looks like there's some non-trivial logic here that we're duplicating. Perhaps we could have a function 
```c
int msp_generate_tract_length(self, double * ret_tract_length)
```
to do the work?

Reply:
Yes, the current way we simulate gc_left and gc_within means we are generating the tract lengths in the same way. Before the speed up it was different. Will add a function.